src/utils/config_filters.cpp

```diff
@@ -20,11 +20,16 @@
 #include "serialization/string_utils.hpp" // for utils::split
 #include "utils/math.hpp"                 // for in_ranges
 
-bool utils::config_filters::bool_matches_if_present(const config& filter, const config& cfg, const std::string& attribute, bool def)
+bool utils::config_filters::bool_matches_if_present_and_valid(const config& filter, const config& cfg, const std::string& attribute, bool def)
 {
 	if(!filter.has_attribute(attribute)) {
 		return true;
 	}
+	//check if filter is boolean, if not then checking can't be done and return true.
```

Comment:
What is the goal? This comment (or the code below) makes no sense to me.

Show a specific example of an ability and a filter that this is useful for.

Reply:
if someone want check affect_self attribute who is boolean with value true by default but write filtet like that:
`affect_self=myself` how sustem will understand that, check if false then what develloper search another value, or then like here declare filter invalid and don't check attribute.

Reply:
Why would this filter be "understandable"? Where is this documented?

Reply:
I believe this is the first time that a boolean attribute has been checked by a filter, so it seems normal to me to ensure that the filter used uses true/yes or false/no and not other things (the attribute affect_allies is an exception because its absence has a meaning other than true/false)

Reply:
You do not ensure only valid values by simply pretending anything else is not there. That's terrible. Raise an error and let the WML author fix it.

Reply:
ok,, i done that.